.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
```

Comment:
Maybe seperate the test step and only have the timeout there? Just a suggestion. This seems fine for now aswell

Reply:
I've set the build's timeout to 40 minutes and the test step's to 20. Is that okay?